### PR TITLE
[cpu] Factor out BundleSaver

### PR DESCRIPTION
**Pull request by bertmaher**
A step towards #1227.  Saving bundles adds a bunch of stuff to CPUBackend's API, so it's hard to tell what can be safely split out.  It's actually quite nicely self-contained, so this diff is mostly a bunch of code motion.

### CODE REVIEW

**Comment by opti-mix** [comment]
@bertmaher Looking good. Have you checked that our bundle examples still work as expected after this change?

**Comment by bertmaher** [comment]
Ah yes, meant to say so in the description.  I checked resnet50:
```
cd examples/bundles/resnet50/
LOADER=`pwd`/../../../build/Release/bin/image-classifier GLOW_SRC=`pwd`/../../../ make run
```

File: lib/Backends/CPU/CPUBackend.cpp [hunk 1]
```diff
@@ -33,10 +37,10 @@
 #include "llvm/Support/raw_ostream.h"
 
 using namespace glow;
+using llvm::StringRef;
```
Comment (opti-mix) [inline lib/Backends/CPU/CPUBackend.cpp]: You are trying to get rid of `llvm::StringRef` in the other PR, right?

Comment (bertmaher) [reply lib/Backends/CPU/CPUBackend.cpp]: Yup :-).  Will update this one once it lands.

**Pull request merged**